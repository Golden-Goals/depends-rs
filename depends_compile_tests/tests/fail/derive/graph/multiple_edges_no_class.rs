use depends::derives::Graph;
#[allow(unused_imports)]
use examples::*;

#[derive(Graph)]
#[depends(
    digraph Dag {
        comments [label="Comments"];
        posts [label="Posts"];

        comments_posts [label="CommentsPosts"];

        comments -> comments_posts [label="TrackCommentPostIds"];
        posts -> comments_posts [label="TrackCommentPostIds"];
    }
)]
struct Graph1 {}

fn main() {}
