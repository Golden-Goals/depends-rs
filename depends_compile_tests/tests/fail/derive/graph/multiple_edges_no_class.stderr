error: Multiple incoming edges but no `class` attribute.
  --> tests/fail/derive/graph/multiple_edges_no_class.rs:11:24
   |
11 |         comments_posts [label="CommentsPosts"];
   |                        ^^^^^^^^^^^^^^^^^^^^^^^
